ingestion/src/metadata/ingestion/sink/elasticsearch.py

```diff
@@ -223,70 +207,89 @@ def __init__(
             )
 
         if self.config.index_users:
             self._check_or_create_index(
-                self.config.user_index_name, USER_ELASTICSEARCH_INDEX_MAPPING
+                ES_INDEX_MAP[User.__name__], USER_ELASTICSEARCH_INDEX_MAPPING
             )
 
         if self.config.index_teams:
             self._check_or_create_index(
-                self.config.team_index_name, TEAM_ELASTICSEARCH_INDEX_MAPPING
+                ES_INDEX_MAP[Team.__name__], TEAM_ELASTICSEARCH_INDEX_MAPPING
             )
 
         if self.config.index_glossary_terms:
             self._check_or_create_index(
-                self.config.glossary_term_index_name,
+                ES_INDEX_MAP[GlossaryTerm.__name__],
                 GLOSSARY_TERM_ELASTICSEARCH_INDEX_MAPPING,
             )
 
         if self.config.index_mlmodels:
             self._check_or_create_index(
-                self.config.mlmodel_index_name,
+                ES_INDEX_MAP[MlModel.__name__],
                 MLMODEL_ELASTICSEARCH_INDEX_MAPPING,
             )
 
         if self.config.index_tags:
             self._check_or_create_index(
-                self.config.tag_index_name,
+                ES_INDEX_MAP[Tag.__name__],
                 TAG_ELASTICSEARCH_INDEX_MAPPING,
             )
 
         if self.config.index_entity_report_data:
             self._check_or_create_index(
-                self.config.entity_report_data_index_name,
+                ES_INDEX_MAP[ReportData.__name__],
                 ENTITY_REPORT_DATA_INDEX_MAPPING,
             )
 
         if self.config.index_web_analytic_user_activity_report_data:
             self._check_or_create_index(
-                self.config.web_analytic_user_activity_report_data_index_name,
+                ES_INDEX_MAP["web_analytic_user_activity_report"],
                 WEB_ANALYTIC_USER_ACTIVITY_REPORT_DATA_INDEX_MAPPING,
             )
 
         if self.config.index_web_analytic_entity_view_report_data:
             self._check_or_create_index(
-                self.config.web_analytic_entity_view_report_data_name,
+                ES_INDEX_MAP["web_analytic_entity_view_report"],
                 WEB_ANALYTIC_ENTITY_VIEW_REPORT_DATA_INDEX_MAPPING,
             )
 
+        if self.config.index_containers:
+            self._check_or_create_index(
+                ES_INDEX_MAP[Container.__name__],
+                CONTAINER_ELASTICSEARCH_INDEX_MAPPING,
+            )
+
+        if self.config.index_queries:
+            self._check_or_create_index(
+                ES_INDEX_MAP[Query.__name__],
+                QUERY_ELASTICSEARCH_INDEX_MAPPING,
+            )
+
+        # Prepare write record dispatching
+        self._write_record = singledispatch(self._write_record)
+        self._write_record.register(Classification, self._write_classification)
+        self._write_record.register(ReportData, self._write_report_data)
+        self._write_record.register(Policy, self._write_policy)
+
```

Comment:
Should we move these into their own class with a `write()` method, instantite that object `here` dynamically and have `def write_record()` call `self.<object>.write(client,  record)`?

Reply:
it would be the same right? The only difference on handling the process in a class or as functions. I'd rather go the function approach here as there's no other methods we'd need to add in the class